### ISSUE TITLE
com.typesafe.config.ConfigException$NotResolved: called valueType() on value with unresolved substitutions, need to Config#resolve() first

### ISSUE THREAD

**Issue by eduardb** [ktorio/ktor#518]
In a Ktor 0.9.3 application, I want to have different `.conf` files for different environments, where I extend one main `.conf` file. However, I am getting the following exception:

```
Exception in thread "main" com.typesafe.config.ConfigException$NotResolved: called valueType() on value with unresolved substitutions, need to Config#resolve() first, see API docs
    at com.typesafe.config.impl.ConfigDelayedMerge.valueType(ConfigDelayedMerge.java:46)
    at com.typesafe.config.impl.SimpleConfig.hasPath(SimpleConfig.java:96)
    at io.ktor.config.HoconApplicationConfigKt.tryGetString(HoconApplicationConfig.kt:33)
    at io.ktor.server.engine.CommandLineKt$commandLineEnvironment$environment$1.invoke(CommandLine.kt:64)
    at io.ktor.server.engine.CommandLineKt$commandLineEnvironment$environment$1.invoke(CommandLine.kt)
    at io.ktor.server.engine.ApplicationEngineEnvironmentBuilder.build(ApplicationEngineEnvironment.kt:55)
    at io.ktor.server.engine.ApplicationEngineEnvironmentKt.applicationEngineEnvironment(ApplicationEngineEnvironment.kt:38)
    at io.ktor.server.engine.CommandLineKt.commandLineEnvironment(CommandLine.kt:50)
    at io.ktor.server.netty.DevelopmentEngine.main(DevelopmentEngine.kt:13)
    at ApplicationKt.main(application.kt:24)
```

This is my `application.conf` file:

```
ktor {
    deployment {
        port = 8080
        port = ${?PORT}
        watch = [ website ]
    }
    application {
        modules = [ ApplicationKt.module ]
    }
}

ui {
  hotreload {
    enabled = false
    path-prefix = "http://localhost:3000"
  }
}
```

And the one extending it that I want to use:

```
include "application.conf"

ui.hotreload.enabled = true
```

After doing some debugging, I saw that the call to `tryGetString` with path `ktor.deployment.port` for value `${?PORT}` (that is coming from the included file) is not resolved for some reason. Removing the line `port = ${?PORT}` prevents the crash, and so does using `aaplication.conf` directly.

**Comment by cy6erGn0m** [comment]
It should be resolved as per PR #488

**Comment by eduardb** [comment]
I can confirm, 0.9.4.-alpha-2 works as expected.
